lua/nvim-tree/diagnostics.lua

```diff
@@ -12,6 +11,24 @@ local severity_levels = {
   Hint = 4,
 }
 
+---@class DiagStatus
+---@field value integer|nil
+---@field cache_version integer
+
+--- The buffer-severity mappings derived during the last diagnostic list update.
+---@type table
+local BUFFER_SEVERITY = {}
```

Comment:
This is looking good; we have a cache similar to git's.

It'll take me a while to properly review and test this one.